MapboxCoreNavigation/CLLocation.swift

```diff
@@ -114,17 +114,27 @@ extension CLLocation {
             return stepCoordinates
         }
         
-        return nearbyCoordinates
+        return legProgress.nearbyCoordinates
     }
     
+    func projectedDistance(for distanceRemaining: CLLocationDistance, previousSpeeds: [CLLocationSpeed]) -> CLLocationDistance {
+        var defaultDeadReckonDistance = speed * RouteControllerDeadReckoningTimeInterval
+        
+        // If the user is reducing in speed, start to dead reckon less
+        let averageDelta = previousSpeeds.reduce(0, +) / Double(previousSpeeds.count)
+        if averageDelta < 0 {
+            defaultDeadReckonDistance = defaultDeadReckonDistance + averageDelta
```

Comment:
`defaultDeadReckonDistance` is a distance measured in meters. It’s incorrect to add `averageDelta`, which is measured in meters per second.